### DISCUSSION TITLE
Cannot create annotation/highlight with screen reader

### DISCUSSION THREAD

**Discussion by rhoadsstevens** [ManifoldScholar/manifold#3477] [Accessibility]
Priority level 1 (out of 3) from UW Accessible Technologies

WCAG mapping: 2.1.1 Keyboard

STEPS TO REPRODUCE:
1. Select a book (i.e. "I My Recall")
2. Turn on caret browsing (insert + F7)
3. Tab into the text
4. Use NVDA/JAWS and put the SR in interactive mode (the cursor from the caret browsing should move with your arrow keys).
5. Highlight some text.
6. Put the SR in reading mode (arrow keys should read the text).
7. Try and highlight the text.

OBSERVED BEHAVIOR
When you are highlighting the text with screen readers in interactive mode, nothing is being announced to screen reader users as the highlights are being made, but visually the text is highlighted and the context menu opens. When in reading mode with NVDA, the text is announced as selected, but visually the text is not highlighted and the context menu never opens. When in reading mode with JAWS, the text is announced as selected and the text is visually highlighted, but only occasionally the context menu opens upon releasing the modifier keys.

Overall, in interactive mode, SR users are unaware of what text they are highlighting so they cannot create an annotation, and in reading mode the text does not always get highlighted and the context menu does not always open so they still cannot create an annotation.

EXPECTED BEHAVIOR:
SR users should be able to highlight text and create annotations. The exact expected behavior for this is up for discussion in terms of whether SR users should try to highlight in interactive mode (which is expected since they are interacting with the text) or in reading mode (since this seems for feasibly/technically possible at this point).

**Comment by dananjohnson** [comment]
We believe this is resolved now. We addressed issues on our end in [v8](https://manifoldscholar.github.io/manifold-docusaurus/docs/administering/release_notes/v8x#keyboard-annotation) to make annotation and highlighting accessible using VoiceOver or caret browsing. At the time there were still limitations in NVDA and JAWS with text selection being trapped in the virtual buffer and not passed to the DOM (and thus to Manifold), but these limitations appear to now be resolved upstream.
